### PR TITLE
rust-installer & rls: remove exclusion from rustfmt & tidy 

### PR DESCRIPTION
**Pull request by klensy**
<strike>based on #112884</strike>

`rust-installer` and `rls` no longer submodules, but not removed from exclude list for rustfmt and tidy, preventing running fmt and lints on them.

### CODE REVIEW

**Comment by klensy** [comment]
Note for self: add https://github.com/rust-lang/rust/pull/112884#discussion_r1239803766 back

**Comment by ChrisDenton** [comment]
Is rls still maintained?

**Comment by klensy** [comment]
> Is rls still maintained?

I see it here https://rust-lang.github.io/rustup-components-history/ so yes, until removed?

File: src/tools/tidy/src/bins.rs [hunk 1]
```diff
@@ -95,6 +95,15 @@ mod os_impl {
         return true;
     }
 
+    // FIXME: remove this when rust-installer test sh files changed
```
Comment (ozkanonur) [inline src/tools/tidy/src/bins.rs:98]: Some extra details would be helpful.

Comment (klensy) [reply src/tools/tidy/src/bins.rs]: Reworded in `RI_EXCLUSION_LIST`.

**Pull request merged**